browser/net/BUILD.gn

```diff
@@ -80,6 +80,7 @@ source_set("unit_tests") {
     "//brave/browser/net:geolocation",
     "//brave/components/brave_ads/browser",
     "//brave/components/brave_ads/browser:test_support",
+    "//brave/components/brave_ads/core/public:headers",
```

Comment:
we can use "//brave/components/brave_ads/core" here because it already includes "//brave/components/brave_ads/core/public:headers" as a public dependency